src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -188,6 +191,8 @@ public void Dispose()
 
         internal object CaptureDisposable(object service)
         {
+            _captureDisposableCallback?.Invoke(service);
```

Comment:
Is this better than virtual dispatch even when _captureDisposableCallback is null?